src/puppetlabs/trapperkeeper/services/status/status_core.clj

```diff
@@ -98,6 +105,16 @@
     (when (or differing-svc-version? differing-status-version?)
       (throw (IllegalStateException. error-message)))))
 
+(defn validate-protocol
+  "Throws if the protocol is not http or https"
+  [url]
+  (let [protocol (.getProtocol url)]
+    (if-not (contains? #{"http" "https"} protocol)
+      (throw (IllegalStateException.
```

Comment:
`IllegalArgumentException` seems more appropriate than `IllegalStateException` in this case.